Guiao5/Warehouse1_2.java

```diff
@@ -37,27 +37,44 @@ public void supply(String s, int q) {
         }
     }
 
-    // Errado se faltar algum produto...
-    public void consume(String[] a) throws InterruptedException {
-        l.lock();
-        try{
-            int i = 0;
-            while (i < a.length){
-                Product p = get(a[i]);
-                i++;
-                if(p != null){
-                    if (p.q == 0){
-                        p.c.await();
-                        i = 0;
-                    }
-                }
-            }
-            for(String s  : a){
-                get(s).q--;
-            }
-        } finally {
-            l.unlock();
-        }
+
+  // Devolve algum produto que esteja sem stock
+  private Product missing(String[] s){
+    for(String a: s){
+      Product p = get(a);
+      if(p.q==0){
+        return p;
+      }
     }
+    return null;
+  }
```

Comment:
This is the method responsible for checking the availability of each product. Returns the first missing product if it finds any and returns null if all products are available